### PR TITLE
feat(ticket#211): Implement Delete Member Functionality 

### PR DESCRIPTION
**Pull request by Mubashar1009**
## Change Request

shift-incharge or incharge can delete member 

### Implementation

* created `delete-user` action
* used this in `MemberActionRender.tsx` for deleting member 

### Resources

image
image

### Test

image

### CODE REVIEW

File: src/app/members/actions/delete-user.test.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { waitFor } from "@testing-library/react";
+import { describe, expect, it, vi } from "vitest";
+import { deleteMember } from "./delete-user";
+
+describe("Test Delete User Function", () => {
+  it("Delete Function should be called", () => {
```
Comment (Monfernape) [inline src/app/members/actions/delete-user.test.ts:6]: Tests for who can delete member. Also, if I'm not wrong, it's not delete, it's deactivate. cc @muhammad-bilal-ali-khokhar

File: src/app/members/actions/delete-user.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+"use server";
+import { Tables } from "@/lib/db";
+import { getSupabaseClient } from "@/utils/supabase/supabaseClient";
+
+export const deleteMember = async (id: number) => {
+  const supabase = await getSupabaseClient();
+  const response = await supabase.from(Tables.Members).delete().eq("id", id);
```
Comment (Monfernape) [inline src/app/members/actions/delete-user.ts:7]:
```suggestion
  return supabase.from(Tables.Members).delete().eq("id", id);
```

Comment (Monfernape) [reply src/app/members/actions/delete-user.ts:7]: Missing check who can deactive member. Also, let's update the status only.

File: src/app/members/components/MemberActionRender.tsx [hunk 3]
```diff
@@ -24,8 +28,18 @@ const MemberTableActionRender = ({ memberInfo }: Props) => {
     return;
   };
 
-  const handleDeleteMember = () => {
-    return;
+  const handleDeleteMember = async () => {
```
Comment (Monfernape) [inline src/app/members/components/MemberActionRender.tsx:31]: Does list update in realtime when we deactivate a member?

**Review by Monfernape — CHANGES_REQUESTED** [review]
@muhammad-bilal-ali-khokhar members can't be deleted. We just need to update the status, right?

Following up on that, @muhammad-bilal-ali-khokhar let's create a ticket to fetch active users only. For incharge, bring all of them.

File: src/app/members/components/MemberActionRender.tsx [hunk 4]
```diff
@@ -24,8 +25,18 @@ const MemberTableActionRender = ({ memberInfo }: Props) => {
...
+    const result = await deleteMember(id);
+    if (result && result.status === 204) {
```
Comment (muhammad-bilal-ali-khokhar) [inline src/app/members/components/MemberActionRender.tsx:30]: @Mubashar1009 not good approach

File: src/common/ActionMenu/ActionsMenu.tsx [hunk 5]
```diff
@@ -29,6 +29,7 @@ const ActionsMenu = ({ actions }: Props) => {
           variant="ghost"
           size="icon"
           className="focus:outline-none focus-visible:ring-0 active:outline-none active:ring-0"
+          disabled={actions.length === 0 && true}
```
Comment (muhammad-bilal-ali-khokhar) [inline src/common/ActionMenu/ActionsMenu.tsx:32]: @Mubashar1009 
```suggestion
          disabled={!actions.length}
```

File: src/constant/constant.ts [hunk 6]
```diff
@@ -23,6 +23,7 @@ export enum UserStatus {
   Active = "active",
   Inactive = "inactive",
   Invited = "invited",
+  Deactivate = "deactivate",
```
Comment (nadirhussain786) [inline src/constant/constant.ts:26]: @Mubashar1009 We should also consider the potential side effects of adding a new key entries in existing enum.

File: src/types/index.tsx [hunk 7]
```diff
@@ -1,13 +1,16 @@
-export type UserRole = "incharge" | "member" | "shift-incharge";
+import { Shift } from "@/constant/constant";
+
+export type UserRole = "incharge" | "member" | "shift_incharge";
 export type Shifts = "A" | "B" | "C" | "D";
 export type UserStatus = "active" | "inactive" | "pending";
```
Comment (nadirhussain786) [inline src/types/index.tsx:5]: @Mubashar1009  why not add type for `deactivated` user?

**Review by nadirhussain786 — CHANGES_REQUESTED** [review]

File: src/constant/constant.ts (hunk 6 above)
Comment (Monfernape) [reply src/constant/constant.ts:26]: This should be called `deactivated`

Comment (Monfernape) [reply src/constant/constant.ts:26]: Why do we have User status as both enum and type. This is confusing.

**Review by Monfernape — APPROVED** [review]

**Pull request merged**